lib/widgets/emoji_reaction.dart

```diff
@@ -360,3 +366,225 @@ class _TextEmoji extends StatelessWidget {
+
+    if (_viewModel != null) {
+      _controller.removeListener(_handleViewModelUpdate);
+      _viewModel!.dispose();
+      _resultsToDisplay = const [];
+    }
+    _viewModel = EmojiAutocompleteView.init(store: store, query: EmojiAutocompleteQuery(''))
+      ..addListener(_handleViewModelUpdate);
+    _controller.addListener(_handleControllerUpdate);
+  }
+
+  void _handleViewModelUpdate() {
+    if (!mounted) return;
+    setState(() {
+      _resultsToDisplay = List.unmodifiable(_viewModel!.results);
+    });
+  }
+
+  void _handleControllerUpdate() {
+    _viewModel!.query = EmojiAutocompleteQuery(_controller.text);
+  }
+
+  @override
+  void dispose() {
+    _viewModel?.dispose();
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final designVariables = DesignVariables.of(context);
+
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+      Padding(padding: const EdgeInsets.only(left: 8, top: 8),
+        child: Row(children: [
+          Flexible(child: TextField(
+            controller: _controller,
+            autofocus: true,
+            decoration: InputDecoration(
+              hintText: zulipLocalizations.emojiPickerSearchEmoji,
+              contentPadding: const EdgeInsets.only(left: 10, top: 6),
+              filled: true,
+              fillColor: designVariables.bgSearchInput,
+              border: OutlineInputBorder(
+                borderRadius: BorderRadius.circular(10),
+                borderSide: BorderSide.none)),
+            style: const TextStyle(fontSize: 19, height: 26 / 19)
+              .merge(weightVariableTextStyle(context, wght: 400)))),
+          TextButton(
+            onPressed: () => Navigator.pop(context),
+            style: TextButton.styleFrom(
+              padding: EdgeInsets.zero,
+              splashFactory: NoSplash.splashFactory,
+              foregroundColor: designVariables.contextMenuItemText,
+            ).copyWith(backgroundColor: WidgetStateColor.resolveWith((states) =>
+              states.contains(WidgetState.pressed)
+                ? designVariables.contextMenuItemBg.withFadedAlpha(0.20)
+                : Colors.transparent)),
+            child: Text(zulipLocalizations.dialogClose,
+              style: const TextStyle(fontSize: 20, height: 30 / 20)
+                .merge(weightVariableTextStyle(context, wght: 400)))),
+        ])),
+      Expanded(child: InsetShadowBox(
+        top: 8, bottom: 8,
+        color: designVariables.bgContextMenu,
+        child: ListView.builder(
+          padding: const EdgeInsets.all(8),
+          itemCount: _resultsToDisplay.length,
+          itemBuilder: (context, i) => EmojiPickerListEntry(
+            pageContext: widget.pageContext,
+            emoji: _resultsToDisplay[i].candidate,
+            message: widget.message)))),
```

Comment:
The touch feedback is colored by a Material default, `overlayColor`:

<img width="421" alt="image" src="https://github.com/user-attachments/assets/1f11471e-5c28-4cd6-b51f-b4842e1557ae">

I see we don't have a spec for these items' touch feedback in the Figma, but is there something similar there you can infer a good color from?

Also, to me it seems like the touch feedback should extend out to the edge of the bottom action sheet.

Reply:
Also, if the user has already reacted with one of the emoji in the results, how about showing that state by marking the item somehow? Maybe similarly (or the same) as the on-press feedback, would that be reasonable?